vm/map.c

```diff
@@ -1120,6 +1122,32 @@ int vm_mapBelongs(const struct _process_t *proc, const void *ptr, size_t size)
 }
 
 
+int vm_mapStringBelongs(const struct _process_t *proc, const char *str)
```

Comment:
This check has one more vulnerability than `vm_mapBelongs` has. User can also alter the string post NUL byte check. We either have to copy the string or pass the string with length found by this function.